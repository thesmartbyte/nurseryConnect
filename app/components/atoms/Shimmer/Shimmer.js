

import React, { Component } from 'react';
import ContentLoader from 'react-content-loader';
import COLORS from '../../../styles/theme';

const props = {
  speed: 1,
  primaryColor: COLORS.GLOBAL_BACKGROUND_COLOR,
  secondaryColor: COLORS.LIGHT_GREY_3,
};

class Shimmer extends Component {
  render() {
    return (
      <ContentLoader
        speed={props.speed}
        primaryColor={props.primaryColor}
        secondaryColor={props.secondaryColor}
      >
        <rect x="0" y="0" rx="10" ry="10" width="100%" />
      </ContentLoader>
    );
  }
}

export default Shimmer;
