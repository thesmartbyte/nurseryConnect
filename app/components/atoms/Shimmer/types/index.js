

export type Props = {
  speed: number,
  primaryColor: string,
  secondaryColor: string,
}
