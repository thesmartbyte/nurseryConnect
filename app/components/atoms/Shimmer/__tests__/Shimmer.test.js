import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import Shimmer from '../Shimmer';

describe('Shimmer Test', () => {
  test('Should Render Correctly', () => {
    const homePage = shallow(<Shimmer />);
    expect(homePage).toMatchSnapshot();
  });
});
