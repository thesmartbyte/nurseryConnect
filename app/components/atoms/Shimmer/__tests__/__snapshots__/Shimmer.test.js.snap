// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shimmer Test Should Render Correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Shimmer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <rect
        rx="10"
        ry="10"
        width="100%"
        x="0"
        y="0"
      />,
      "primaryColor": "rgba(242, 242, 242, 1)",
      "secondaryColor": "rgba(128,128,128, 1)",
      "speed": 1,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "rx": "10",
        "ry": "10",
        "width": "100%",
        "x": "0",
        "y": "0",
      },
      "ref": null,
      "rendered": null,
      "type": "rect",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <rect
          rx="10"
          ry="10"
          width="100%"
          x="0"
          y="0"
        />,
        "primaryColor": "rgba(242, 242, 242, 1)",
        "secondaryColor": "rgba(128,128,128, 1)",
        "speed": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "rx": "10",
          "ry": "10",
          "width": "100%",
          "x": "0",
          "y": "0",
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
